packages/plugin-core/src/views/CalendarView.ts

```diff
@@ -0,0 +1,141 @@
+        // This would result in the first call to unset `noteActive` and the seconde one to set it again. This might create flashing UI changes that disturb the eye.
+        _.debounce(this.handleActiveTextEditorChange, 100),
+        this
+      )
+    );
+    dendronWorkspace.addDisposable(
+      vscode.workspace.onDidSaveTextDocument(this.handleSaveTextDocument, this)
+    );
+  }
+
+  openTextDocument(document: vscode.TextDocument) {
+    const ctx = "CalendarView:openTextDocument";
+    if (!getWS().workspaceService?.isPathInWorkspace(document.uri.fsPath)) {
+      Logger.info({
+        ctx,
+        uri: document.uri.fsPath,
+        msg: "not in workspace",
+      });
+      return;
+    }
+    const note = VSCodeUtils.getNoteFromDocument(document);
+    if (note) {
+      Logger.info({
+        ctx,
+        msg: "refresh note",
+        note: NoteUtils.toLogObj(note),
+      });
+      this.refresh(note);
+    }
+  }
+
+  async handleSaveTextDocument(document: vscode.TextDocument) {
+    this.openTextDocument(document); // TODO optimize for only daily notes
+  }
+
+  async handleActiveTextEditorChange() {
+    const document = VSCodeUtils.getActiveTextEditor()?.document;
+    if (document) {
+      this.openTextDocument(document);
+    } else {
+      this.refresh(); // call refresh without note so that `noteActive` gets unset.
+    }
+  }
+
+  public postMessage(msg: DMessage) {
+    this._view?.webview.postMessage(msg);
+  }
+
+  public resolveWebviewView(
+    webviewView: vscode.WebviewView,
+    _context: vscode.WebviewViewResolveContext,
+    _token: vscode.CancellationToken
+  ) {
+    const ctx = "CalendarView:resolveWebView";
+    this._view = webviewView;
+    let start = process.hrtime();
+    Logger.info({ ctx, msg: "enter", start });
+    webviewView.webview.options = {
+      enableScripts: true,
+      localResourceRoots: [],
+    };
+    webviewView.webview.html = this._getHtmlForWebview(webviewView.webview);
+    webviewView.webview.onDidReceiveMessage(
+      async (msg: CalendarViewMessage) => {
+        Logger.info({ ctx: "onDidReceiveMessage", data: msg });
+        switch (msg.type) {
+          case CalendarViewMessageType.onSelect: {
+            console.log("onDidReceiveMessage:onSelect:data", msg.data);
+            if (msg.data.id) {
+              const note = getEngine().notes[msg.data.id];
+              await new GotoNoteCommand().execute({
+                qs: note.fname,
+                vault: note.vault,
+              });
+            } else if (msg.data.fname) {
+              await new CreateDailyJournalCommand().execute({
+                fname: msg.data.fname,
+              });
+            }
+            break;
+          }
+          case CalendarViewMessageType.onGetActiveEditor: {
+            this.handleActiveTextEditorChange(); // initial call
+            break;
+          }
+          default:
+            console.log("got data", msg);
```

Comment:
do we get any other messages? this should throw an error if we don't. see [this](https://wiki.dendron.so/notes/773e0b5a-510f-4c21-acf4-2d1ab3ed741e.html#prefer-compile-time-checks-for-exhaustive-patterns)

Reply:
a20c5b12